sdk/core/azure-core/src/main/java/com/azure/core/http/policy/UserAgentPolicy.java

```diff
@@ -62,82 +71,113 @@ public UserAgentPolicy(String userAgent) {
-     * in the user agent.
+     * <p>If the passed configuration contains true for AZURE_TELEMETRY_DISABLED the platform information won't be
+     * included in the user agent.</p>
      *
+     * @deprecated This method has been deprecated in favor of {@link
+     * #UserAgentPolicy(String, String, String, Configuration)}.
      * @param sdkName Name of the client library.
      * @param sdkVersion Version of the client library.
      * @param version {@link ServiceVersion} of the service to be used when making requests.
-     * @param configuration Configuration store that will be checked for the AZURE_TELEMETRY_DISABLED.
-     * @throws NullPointerException if {@code configuration} is {@code null}.
+     * @param configuration Configuration store that will be checked for {@link
+     * Configuration#PROPERTY_AZURE_TELEMETRY_DISABLED}. If {@code null} is passed the {@link
+     * Configuration#getGlobalConfiguration() global configuration} will be checked.
      */
+    @Deprecated
     public UserAgentPolicy(String sdkName, String sdkVersion, Configuration configuration, ServiceVersion version) {
-        Objects.requireNonNull(configuration, "'configuration' cannot be null.");
-        boolean telemetryDisabled = configuration.get(Configuration.PROPERTY_AZURE_TELEMETRY_DISABLED, false);
-        if (telemetryDisabled) {
-            this.userAgent = String.format(DISABLED_TELEMETRY_USER_AGENT_FORMAT, sdkName, sdkVersion,
-                version.getVersion());
-        } else {
-            this.userAgent = String.format(USER_AGENT_FORMAT, sdkName, sdkVersion, getPlatformInfo(),
-                version.getVersion());
-        }
+        this.userAgent = buildUserAgent(null, sdkName, sdkVersion, configuration);
     }
 
     /**
      * Updates the "User-Agent" header with the value supplied in the policy.
      *
-     * When the User-Agent header already has a value and it differs from the value used to create this policy the
-     * User-Agent header is updated by prepending the value in this policy.
-     * {@inheritDoc}
+     * <p>The {@code context} will be checked for {@code Override-User-Agent} and {@code Append-User-Agent}.
+     * {@code Override-User-Agent} will take precedence over the value supplied in the policy,
+     * {@code Append-User-Agent} will be appended to the value supplied in the policy.</p>
+     *
+     * @param context request context
+     * @param next The next policy to invoke.
+     * @return A publisher that initiates the request upon subscription and emits a response on completion.
      */
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
-        String header = context.getHttpRequest().getHeaders().getValue("User-Agent");
-        if (header == null || header.contains(DEFAULT_USER_AGENT_HEADER)) {
-            header = userAgent;
+        String overrideUserAgent = (String) context.getData(OVERRIDE_USER_AGENT_CONTEXT_KEY).orElse(null);
```

Comment:
We should also consider what happens to the user agent strings that were generated with the released version of the API and how this change will impact the user data analytics. Do we have other languages also making these changes to the user agent string?

Reply:
I'll search the other languages handling on this, if need be we can revert this portion of the change.